crates/nu-command/src/filesystem/uutil-notes.md

```diff
@@ -0,0 +1,120 @@
+# Using uu_cratename in nushell
+
+## Argument Passing
+
+We need to get arguments form nushell and pass arguments to uumain. I seriously doubt that the nushell argument structures will be 100% compatible with clap, which is what is used in uutils, so there could be some rough edges.
+
+This is a prototype for copy in PR #9463
+```rust
+fn run(
+    &self,
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+    _input: PipelineData,
+) -> Result<PipelineData, ShellError> {
+    // Get the App
+    // let mut app = uu_cp::uu_app();
+    // app.print_help()?;
+
+    // Create uucore::Args somehow from nushell args
+    let s1 = "cp".to_string();
+    let s2 = "-h".to_string();
+    let args = vec![OsString::from(s1), OsString::from(s2)];
+    // Pass uucore::Args to app.uumain
+    uu_cp::uumain(&mut args.into_iter());
```

Comment:
how does uumain read in stdin? Can we redirect it?

Reply:
This is the first few lines from `uumain`
```rust
#[uucore::main]
pub fn uumain(args: impl uucore::Args) -> UResult<()> {
    let matches = uu_app().try_get_matches_from(args);

```
So, it's reading the args passed in.